libbeat/publisher/queue/diskqueue/encryption_test.go

```diff
@@ -0,0 +1,57 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package diskqueue
+
+import (
+	"bytes"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEncryptionRoundTrip(t *testing.T) {
+	tests := map[string]struct {
+		plaintext []byte
+	}{
+		"8 bits":   {plaintext: []byte("a")},
+		"128 bits": {plaintext: []byte("bbbbbbbbbbbbbbbb")},
+		"136 bits": {plaintext: []byte("ccccccccccccccccc")},
+	}
+	for name, tc := range tests {
+		pr, pw := io.Pipe()
+		src := bytes.NewReader(tc.plaintext)
+		var dst bytes.Buffer
+		key := []byte("kkkkkkkkkkkkkkkk")
+
+		go func() {
+			//NewEncryptionWriter writes iv, so needs to be in go routine
+			ew, err := NewEncryptionWriter(pw, key)
+			assert.Nil(t, err, name)
+			_, err = io.Copy(ew, src)
+			assert.Nil(t, err, name)
+			ew.Close()
```

Comment:
Should there be a check at the end here that the encrypted result is not equal to the plaintext? I think this test might still pass if these functions did not encrypt anything.

Reply:
Yeah, that's a problem with round trip tests.

I added a TeeReader, and check that the lenght of the iv + length of the plaintext gets written and that the plaintext & ciphertext don't match.